dotnet/test/common/RelativeLocatorTest.cs

```diff
@@ -53,14 +75,166 @@ public void ShouldBeAbleToCombineFilters()
+            var values = seen.Select(element => element.GetDomAttribute("id"));
+            Assert.That(values, Is.EquivalentTo(new List<string> { "second", "eighth", "fourth", "sixth", "first", "third", "seventh", "ninth" }));
+        }
+
+        [Test]
+        public void ExerciseNearLocatorWithXpath()
+        {
+            driver.Url = (EnvironmentManager.Instance.UrlBuilder.WhereIs("relative_locators.html"));
+
+            ReadOnlyCollection<IWebElement> seen = driver.FindElements(RelativeBy.WithLocator(By.XPath("//td")).Near(By.Id("center")));
+
+            // Elements are sorted by proximity and then DOM insertion order.
+            // Proximity is determined using distance from center points, so
+            // we expect the order to be:
+            // 1. Directly above (short vertical distance, first in DOM)
+            // 2. Directly below (short vertical distance, later in DOM)
+            // 3. Directly left (slight longer distance horizontally, first in DOM)
+            // 4. Directly right (slight longer distance horizontally, later in DOM)
+            // 5-8. Diagonally close (pythagoras sorting, with top row first
+            //    because of DOM insertion order)
+            var values = seen.Select(element => element.GetDomAttribute("id"));
+            Assert.That(values, Is.EquivalentTo(new List<string> { "second", "eighth", "fourth", "sixth", "first", "third", "seventh", "ninth" }));
+        }
+
+        [Test]
+        public void ExerciseNearLocatorWithCssSelector()
+        {
+            driver.Url = (EnvironmentManager.Instance.UrlBuilder.WhereIs("relative_locators.html"));
+
+            ReadOnlyCollection<IWebElement> seen = driver.FindElements(RelativeBy.WithLocator(By.CssSelector("td")).Near(By.Id("center")));
+
+            // Elements are sorted by proximity and then DOM insertion order.
+            // Proximity is determined using distance from center points, so
+            // we expect the order to be:
+            // 1. Directly above (short vertical distance, first in DOM)
+            // 2. Directly below (short vertical distance, later in DOM)
+            // 3. Directly left (slight longer distance horizontally, first in DOM)
+            // 4. Directly right (slight longer distance horizontally, later in DOM)
+            // 5-8. Diagonally close (pythagoras sorting, with top row first
+            //    because of DOM insertion order)
+            var values = seen.Select(element => element.GetDomAttribute("id"));
+            Assert.That(values, Is.EquivalentTo(new List<string> { "second", "eighth", "fourth", "sixth", "first", "third", "seventh", "ninth" }));
+        }
+
+        [Test]
+        public void EnsureNoRepeatedElements()
+        {
+            driver.Url = EnvironmentManager.Instance.UrlBuilder.CreateInlinePage(new InlinePage()
+               .WithTitle("Repeated Elements")
+               .WithStyles(
+                " .c {\n"
```

Comment:
It is better to use row string literals: https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/tokens/raw-string